import "soc_status.proto";
import "position_status.proto";
import "vehicle_status.proto";

option java_package = "com.nextev.cvs_proto.protobuf.events";
option java_outer_classname = "ChargeEndMessage";

message ChargeEndEvent {

    optional string id = 1; // unique id, VIN for vehicle data
    optional int32 version = 2; // message protocol version
    optional string charge_id = 3; // ChargeID = YYMMDD####
    optional uint64 sample_ts = 4; // time when charging ends
    optional SOCStatus soc_status = 5; // soc status
    optional SOCStatus.ChargingInfo charging_info = 6; // charging info
    optional PositionStatus position_status = 7;
    optional VehicleStatus vehicle_status = 8;
}